addons/mail/models/res_partner.py

```diff
@@ -130,7 +130,7 @@ def get_mention_suggestions(self, search, limit=8, channel_id=None):
             If channel_id is given, only members of this channel are returned.
         """
         search_dom = expression.OR([[('name', 'ilike', search)], [('email', 'ilike', search)]])
-        search_dom = expression.AND([[('active', '=', True)], search_dom])
+        search_dom = expression.AND([[('active', '=', True), ('type', '!=', 'private')], search_dom])
```

Comment:
Note: this just serves as not returning private partners, but the JS might still suggest them if they are known through other means.

A complete solution should include the JS being aware of this information (through `mail_partner_format` return value + corresponding `convertData` and field in JS model), and then a corresponding filter in `searchSuggestions`.